### ISSUE TITLE
How can I calculate FLOPs /MACs and Params without 0 weights?

### ISSUE THREAD

**Issue by Boltzmachine** [Lyken17/pytorch-OpCounter#129]
I prune the model torch.nn.utils.prune. It will modify the pruned weights to 0s. So how can I use thop to calculate without these zeros

**Comment by Lyken17** [comment]
Can you show some examples about how you prune the network?

**Comment by AndreaBrg** [comment]
@Lyken17 I'm facing the same issue. I think it would be nice for thop to support the calculation of MACs for pruned neuron (i.e neurons which weights are all zeros). Due to the popularity of (structured) pruning procedure this could allow for a theoretical calculation of MACs reduction (dense network vs pruned network).

I tried a custom implementation of the `count_linear` hook, if you think this is useful/interesting I could try building a PR.
Here the implementation:

```
def count_linear(m, x, y):
    total_mul = m.in_features
    num_elements = m.out_features
    
    zeroed_neurons = num_elements - torch.nonzero(torch.abs(m.weight).sum(dim=1)).shape[0]  # Get the zeroed neurons
    
    total_ops = total_mul * (num_elements - zeroed_neurons)
    
    m.total_ops += torch.DoubleTensor([int(total_ops)])
```

**Comment by Lyken17** [comment]
@AndreaBrg That would be a great feature to have. However, is there a stable api for pruning in pytorch?  I see the torch.prune utiliies is still under construction and the api might change in the future.

**Comment by AndreaBrg** [comment]
Yes, torch.prune is the official api for pruning a torch model, but usually you can just apply some kind of threshold which will set a set of parameters to zero, ingoring the torch api.

Something like:
`p.copy_(torch.where(torch.abs(p) < T, torch.zeros_like(p), p))`
where p is a parameter (e.g. a layer .weight attribute).

**Comment by Ashutosh1995** [comment]
Is there any update on this?
